workflow/validate/validate.go

```diff
@@ -282,6 +282,21 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	if err != nil {
 		return err
 	}
+	// validate hook if it is an exit hook
+	for hookName, hook := range wf.Spec.Hooks {
+		if hookName != wfv1.ExitLifecycleEvent {
+			continue
+		}
+		tmplHolder = &wfv1.WorkflowStep{Template: hook.Template}
+		if hook.TemplateRef != nil {
+			tmplHolder = &wfv1.WorkflowStep{TemplateRef: hook.TemplateRef}
+		}
+		ctx.globalParams[common.GlobalVarWorkflowFailures] = placeholderGenerator.NextPlaceholder()
+		_, err = ctx.validateTemplateHolder(tmplHolder, tmplCtx, &hook.Arguments, opts.WorkflowTemplateValidation)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I think we better have this as part of `validateHooks` function.
So we validates the hooks at both the workflow and template level

Reply:
I think validating the exit hook at both template level and workflow level makes sense. However, `workflow.failures` is not on the scope of a step's exit hook, so the validation should fail, correct?

Reply:
Yes, it should fail. 

There is an edge case of the last template of a Workflow, which could technically be equivalent to a Workflow-level exit hook, but I think we can more-or-less ignore that scenario as the user intent is not there